### DISCUSSION TITLE
How can i add a custom filter function for jinjia?

### DISCUSSION THREAD

**Discussion by JanEricNitschkeAA** [copier-org/copier#1895] [Q&A]
Specifically i am prompting the user to specify the python versions they want to support (via poetry syntax) and want to extract the minimum support version using the "https://pypi.org/project/packaging/" package (+ some custom logic).

Current approach is to use https://pypi.org/project/copier-templates-extensions/ but i dont want the users to have to run additional commands (brew for copier install, pip for extension install and then invoke copier)

**Comment by pawamoy** [comment]
One-liner for your users:

```
uvx copier --with copier-templates-extensions copy --trust gh:your/template their-project
```

The uvx command comes from the `uv` package, installable with curl or from pypi :)

**Reply by JanEricNitschkeAA** [reply]
So i would first install uv, then copier with uvx and then i can instantiate the template with the extensions with that command?

**Reply by pawamoy** [reply]
Yes. you install [uv](https://github.com/astral-sh/uv) once, then can use `uvx` to run tools like Copier, with additional dependencies like copier-templates-extension if needed. `uvx` is just a shortcut for `uv tool run`. You could also install Copier first, then run it with:

```bash
uv tool install copier --with copier-templates-extensions
uv tool run copier copy --trust gh:your/template their-project
```
